labs/a10.md

```diff
@@ -0,0 +1,371 @@
+this point in the course, I trust you can figure out how to install it just given that link 🙂 _hint: use `wget` and`dpkg`_
+
+After installing, I recommend running the command `sudo usermod -aG docker $USER` so
+you can use Docker as a non-root user. This means you won't have to type `sudo docker` all the time. This is optional
+but for the rest of this exercise I'm going to assume that you did this. If you see some output like
+
+```
+sent invalidate(passwd) request, exiting
+sent invalidate(group) request, exiting
+```
+
+This is normal, it's just adding a user to a group.
+
+### Creating your first Docker container
+To verify that you installed things correctly, try running
+
+`docker run hello-world`
+
+You should see some friendly output like so (hashes are probably different, don't worry about it):
+
+```
+Unable to find image 'hello-world:latest' locally
+latest: Pulling from library/hello-world
+b04784fba78d: Already exists
+Digest: sha256:f3b3b28a45160805bb16542c9531888519430e9e6d6ffc09d72261b0d26ff74f
+Status: Downloaded newer image for hello-world:latest
+
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+
+To generate this message, Docker took the following steps:
+ 1. The Docker client contacted the Docker daemon.
+ 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
+ 3. The Docker daemon created a new container from that image which runs the
+    executable that produces the output you are currently reading.
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+
+To try something more ambitious, you can run an Ubuntu container with:
+ $ docker run -it ubuntu bash
+
+Share images, automate workflows, and more with a free Docker ID:
+ https://cloud.docker.com/
+
+For more examples and ideas, visit:
+ https://docs.docker.com/engine/userguide/
+ ```
+
+Some [quick definitions from Docker's website:](https://docs.docker.com/get-started/#a-brief-explanation-of-containers)
+
+An **image** is a lightweight, stand-alone, executable package that includes everything needed to run a piece of
+software, including the code, a runtime, libraries, environment variables, and config files. Images are useful primarily for their
+speed, but images can also be used as a base to be built on top of in future images, as you'll see later with Dockerfiles. In the
+last example hello-world was the image used to test our docker installation.
+
+A **container** is a runtime instance of an image—what the image becomes in memory when actually executed.
+It runs completely isolated from the host environment by default, only accessing host files and ports if configured to
+do so. A container gets created upon executing docker run on an image.
+
+This is similar to the distinction between Objects and Classes in Object Oriented Programming. Images would be Classes,
```

Comment:
I don't think you have to capitalize all of this, it just seems unnecessary to me, but either capitalized or lowercase seems fine to me.